tests/integration/test_package_managers.py

```diff
@@ -111,45 +110,9 @@ def test_packages(
         test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
     )
 
-    output_folder = os.path.join(tmpdir, f"{test_case}-output")
-    cmd = [
-        "fetch-deps",
-        "--source",
-        source_folder,
-        "--output",
-        output_folder,
-    ]
-    if test_params.flags:
-        cmd += test_params.flags
-
-    cmd.append(json.dumps(test_params.packages).encode("utf-8"))
-
-    (output, rc) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
-    assert rc == test_params.expected_rc, (
-        f"Fetching deps ended with unexpected exitcode: {rc} != {test_params.expected_rc}, "
-        f"output-cmd: {output}"
+    _ = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
     )
-    assert test_params.expected_output in str(
-        output
-    ), f"Expected msg {test_params.expected_output} was not found in cmd output: {output}"
-
-    if test_params.check_output_json:
-        output_json = utils.load_json(os.path.join(output_folder, "output.json"))
-        expected_output_json = utils.load_json(
-            os.path.join(test_data_dir, test_case, "output.json")
-        )
-        log.info("Compare output.json files")
-        assert output_json == expected_output_json
-
-    if test_params.check_deps_checksums:
-        files_checksums = utils.calculate_files_sha256sum_in_dir(
-            os.path.join(output_folder, "deps")
-        )
-        expected_files_checksums = utils.load_json(
-            os.path.join(test_data_dir, test_case, "fetch_deps_sha256sums.json")
-        )
-        log.info("Compare checksums of fetched deps files")
-        assert files_checksums == expected_files_checksums
 
     if test_params.check_vendor_checksums:
```

Comment:
Shouldn't the vendor check also be part of `fetch_deps_and_check_output()`?

Reply:
sure, moved to fetch_deps_and_check_output